rotkehlchen/api/v1/encoding.py

```diff
@@ -400,6 +400,39 @@ def _deserialize(
         return asset
 
 
+class MaybeAssetField(fields.Field):
+
+    def __init__(self, *, form_with_incomplete_data: bool = False, **kwargs: Any) -> None:  # noqa: E501
+        self.form_with_incomplete_data = form_with_incomplete_data
+        super().__init__(**kwargs)
+
+    @staticmethod
+    def _serialize(
+            value: Asset,
+            attr: str,  # pylint: disable=unused-argument
+            obj: Any,  # pylint: disable=unused-argument
+            **_kwargs: Any,
+    ) -> Optional[str]:
+        # Asset can be missing so we need to handle None when serializing from schema
+        return str(value.identifier) if value else None
+
+    def _deserialize(
+            self,
+            value: str,
+            attr: Optional[str],  # pylint: disable=unused-argument
+            data: Optional[Mapping[str, Any]],  # pylint: disable=unused-argument
+            **_kwargs: Any,
+    ) -> Optional[Asset]:
+        try:
+            asset = Asset(value, form_with_incomplete_data=self.form_with_incomplete_data)
+        except DeserializationError as e:
+            raise ValidationError(str(e)) from e
+        except UnknownAsset:
+            log.error(f'Failed to deserialize asset {value}')
```

Comment:
Does this solution really work? Wouldn't it just create a list of assets and `None` which would end up being fed to https://github.com/rotki/rotki/blob/c3c9f57b51d7bd1988d4d17cd6d3cc09284e9d63/rotkehlchen/api/rest.py#L406?

Ah wait I just saw your change in v1/resources.py .... hmm ... I don't really like this.

Or well at least  add some more info on the the log entry. That a non-supported asset was passed to the API.